FreeRTOS-Plus/Demo/AWS/Device_Defender_Windows_Simulator/Device_Defender_Demo/DemoTasks/DefenderDemoExample.c

```diff
@@ -0,0 +1,684 @@
+                                   "ulReportId",
+                                   sizeof( "ulReportId" ) - 1,
+                                   &( ucReportIdString ),
+                                   &( xReportIdStringLength ) );
+
+        if( eJsonResult != JSONSuccess )
+        {
+            LogError( ( "ulReportId key not found in the response from the"
+                        "AWS IoT Device Defender Service: %.*s.",
+                        ( int ) ulDefenderResponseLength,
+                        pcDefenderResponse ) );
+        }
+    }
+
+    if( eJsonResult == JSONSuccess )
+    {
+        ulReportIdInResponse = ( uint32_t ) strtoul( ucReportIdString, NULL, 10 );
+
+        /* Is the ulReportId present in the response same as was sent in the
+         * published report? */
+        if( ulReportIdInResponse == ulReportId )
+        {
+            LogInfo( ( "A valid reponse with ulReportId %u received from the "
+                       "AWS IoT Device Defender Service.", ulReportId ) );
+            xStatus = true;
+        }
+        else
+        {
+            LogError( ( "Unexpected ulReportId found in the response from the AWS"
+                        "IoT Device Defender Service. Expected: %u, Found: %u, "
+                        "Complete Response: %.*s.",
+                        ulReportIdInResponse,
+                        ulReportId,
+                        ( int ) ulDefenderResponseLength,
+                        pcDefenderResponse ) );
+        }
+    }
+
+    return xStatus;
+}
+/*-----------------------------------------------------------*/
+
+static void prvPublishCallback( MQTTContext_t * pxMqttContext,
+                                MQTTPacketInfo_t * pxPacketInfo,
+                                MQTTDeserializedInfo_t * pxDeserializedInfo )
+{
+    uint16_t usPacketIdentifier;
+
+    configASSERT( pxMqttContext != NULL );
+    configASSERT( pxPacketInfo != NULL );
+    configASSERT( pxDeserializedInfo != NULL );
+
+    /* Suppress the unused parameter warning when asserts are disabled in
+     * build. */
+    ( void ) pxMqttContext;
+
+    usPacketIdentifier = pxDeserializedInfo->packetIdentifier;
+
+    /* Handle an incoming publish. The lower 4 bits of the publish packet
+     * type is used for the dup, QoS, and retain flags. Hence masking
+     * out the lower bits to check if the packet is publish. */
+    if( ( pxPacketInfo->type & 0xF0U ) == MQTT_PACKET_TYPE_PUBLISH )
+    {
+        configASSERT( pxDeserializedInfo->pPublishInfo != NULL );
+
+        /* Invoke the application callback for incoming publishes. */
+        DefenderStatus_t status;
+        DefenderTopic_t api;
+        bool validationResult;
+        MQTTPublishInfo_t * pPublishInfo = pxDeserializedInfo->pPublishInfo;
+
+        /* Silence compiler warnings about unused variables. */
+        ( void ) usPacketIdentifier;
+
+        status = Defender_MatchTopic( pPublishInfo->pTopicName,
```

Comment:
Adding a few comments in this step will help to improve the readability of this demo.